deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java

```diff
@@ -204,13 +204,21 @@ public Response create(
         EnvironBean envBean = Utils.getEnvStage(environDAO, envName, stageName);
         authorizer.authorize(sc, new Resource(envBean.getEnv_name(), Resource.Type.ENV), Role.OPERATOR);
         String operator = sc.getUserPrincipal().getName();
+
         if(StringUtils.isEmpty(buildId)) {
             throw new TeletaanInternalException(Response.Status.BAD_REQUEST, "Build id can not be empty.");
         }
         BuildBean buildBean = buildDAO.getById(buildId);
+        
+        // check build name must match stage config
         if(!buildBean.getBuild_name().equals(envBean.getBuild_name())) {
             throw new TeletaanInternalException(Response.Status.BAD_REQUEST, String.format("Build name (%s) does not match stage config (%s).", buildBean.getBuild_name(), envBean.getBuild_name()));
         }
+        // check if the stage is whitelisted (allow_private_build) for private build
+        if(buildBean.getScm_branch().equals("private") && ! envBean.getAllow_private_build()) {
```

Comment:
One general question here:
How we whitelist stage for customers? 
By setting their branch name to private?

Reply:
do a manual DB update to the allow_private_build column in the "environs" table.